editor/selectors.js

```diff
@@ -37,6 +38,20 @@ export function getEditedPostTitle( state ) {
 		: state.editor.edits.title;
 }
 
+export function getEditedPostPreviewLink( state ) {
+	const link = state.currentPost.link;
+	if ( ! link ) {
+		return null;
+	}
+
+	const queryStringPosition = link.indexOf( '?' );
```

Comment:
This looks fairly complex. Could this be simplified to look like the implementation in `actions.js`, except with `preview` instead of `post_id` ?

https://github.com/WordPress/gutenberg/blob/052a5d6/editor/actions.js#L24-L29

Alternatively, a generic equivalent to [`add_query_arg`](https://developer.wordpress.org/reference/functions/add_query_arg/) sounds like it'd be a nice utility ([related](https://github.com/Automattic/wp-calypso/blob/master/client/lib/route/add-query-args.js)).